core/Access.php

```diff
@@ -605,6 +605,48 @@ public static function doAsSuperUser($function)
 
         return $result;
     }
+
+    /**
+     * Returns the level of access the current user has to the given site.
+     *
+     * @param int $idSite The site to check.
+     * @return string The access level, eg, 'view', 'admin', 'noaccess'.
+     */
+    public function getRoleForSite($idSite)
+    {
+        if ($this->hasSuperUserAccess
+            || in_array($idSite, $this->getSitesIdWithAdminAccess())
+        ) {
+            return 'admin';
+        }
+
+        if (in_array($idSite, $this->getSitesIdWithViewAccess())) {
```

Comment:
This would also need to handle the new `write` role or will this feature be added later?

Reply:
Yes, should have been added here, will add it.